File: src/ts/magma/api/collect/ListCollector.ts
import { List } from "./List";
import { Collector } from "./Collector";
export class ListCollector<T> implements Collector<T, List<T>> {
	createInitial(): List<T> {
		return Lists.empty();
	}
	fold(current: List<T>, element: T): List<T> {
		return current.addLast(element);
	}
}

File: src/java/magma/api/collect/ListCollector.java
package magma.api.collect;

public class ListCollector<T> implements Collector<T, List<T>> {
    @Override
    public List<T> createInitial() {
        return Lists.empty();
    }

    @Override
    public List<T> fold(List<T> current, T element) {
        return current.addLast(element);
    }
}
